pkg/agent/clean/node_notwindows.go

```diff
@@ -0,0 +1,591 @@
+	}
+
+	if _, err := exec.Command("sh", "-c",
+		"iptables-save -t nat | egrep -w \"COMMIT|DOCKER|docker0|\\*nat\" > nat.txt").Output(); err != nil {
+		return err
+	}
+
+	ipt, err := iptables.New()
+	if err != nil {
+		return err
+	}
+
+	for _, table := range []string{"nat", "mangle"} {
+		logrus.Infof("clearing and deleting iptables table: %s", table)
+		if err := ipt.ClearAndDeleteChain(table, ""); err != nil {
+			return err
+		}
+	}
+
+	if err := ipt.ClearAll(); err != nil {
+		return err
+	}
+	if err := ipt.DeleteAll(); err != nil {
+		return err
+	}
+
+	if _, err := exec.Command("sh", "-c",
+		"iptables-restore filter.txt").Output(); err != nil {
+		return fmt.Errorf("error putting back filter table: %s", err)
+	}
+
+	if _, err := exec.Command("sh", "-c",
+		"iptables-restore nat.txt").Output(); err != nil {
+		return fmt.Errorf("error putting back nat table: %s", err)
+	}
+
+	return nil
+}
+
+func getPaths() []string {
+	pathPrefix := getPrefixPath()
+
+	// all paths on the hosts to rm -rf
+	return []string{
+		hosts.ToCleanCNIConf, hosts.ToCleanCNIBin,
+		"/opt/rke", "/run/secrets/kubernetes.io",
+		"/run/calico", "/run/flannel", "/var/log/containers", "/var/log/pods",
+		hosts.ToCleanCalicoRun,
+		path.Join(pathPrefix, hosts.ToCleanSSLDir),
+		path.Join(pathPrefix, hosts.ToCleanTempCertPath),
+		path.Join(pathPrefix, hosts.ToCleanEtcdDir),
+		path.Join(pathPrefix, hosts.ToCleanCNILib),
+		path.Join(pathPrefix, "/var/lib/kubelet"),
+		path.Join(pathPrefix, "/var/lib/calico"),
+		path.Join(pathPrefix, "/var/lib/weave"),
+		path.Join(pathPrefix, "/etc/ceph"),
+	}
+}
+
+func usage() string {
+	return `agent clean usage for linux:
+	Using the clean commands can be run directly from the agent with proper flags, to clean everything you can run the following:
+
+	docker run --privileged --network host -v /:/host -v /var/run/docker.sock:/var/run/docker.sock rancher/rancher-agent:master -- agent clean node
+
+	The above mounts and grants access to everything the node cleaner will need to complete
+	Note: If your cluster was created with a prefixPath use the env param -e PREFIX_PATH=/my/prefix
+
+Running individual cleanup sub commands
+	docker run --privileged -v /:/host rancher/rancher-agent:master -- agent clean paths
+
+commands:
+	node - cleans the entire node and performs all actions below and is the default command, requires docker socket to clean up dockerd
+	links - clean the interface links created on the host, requires --privileged and --network host
+	paths - clean the paths added by the host like /etc/rancher, /var/lib/rancher, etc. requires -v /:/host 
+	firewall - clean the iptables firewall rules, needs --privileged and --network host to work 
+	script - prints a bash script you can run to clean up the node from the cli
+	help - print this help message
+
+other commands for automation:
+	job - used by the k8s batch job to start the ` + NodeCleanupContainerName + ` container to watch kubelet cleanup and wait to clean the node
+`
```

Comment:
These will all print out indented since there is a tab at the beginning.

Reply:
good point.